src/mps/mps.jl

```diff
@@ -585,9 +585,44 @@ psi = randomMPS(s, n->isodd(n) ? "Up" : "Dn"; linkdims=m)
 Cuu = correlation_matrix(psi,"Cdagup","Cup";site_range=2:8)
 ```
 """
-function correlation_matrix(psi::MPS, Op1::AbstractString, Op2::AbstractString; kwargs...)
+function correlation_matrix(psi::MPS, _Op1::AbstractString, _Op2::AbstractString; kwargs...)
   N = length(psi)
   ElT = promote_itensor_eltype(psi)
+  s = siteinds(psi)
+
+  Op1=_Op1 #make copies into which we can insert "F" string operators.
+  Op2=_Op2
+  onsiteOp = "$Op1*$Op2"
+  fermionic1 = has_fermion_string(Op1, s[1])
+  fermionic2 = has_fermion_string(Op2, s[1])
+  if fermionic1!=fermionic2
+    error("correlation_matrix: Mixed fermionic and bosonic operators are not supported yet.")
+  end
+  
+  # Decide if we need to calculate a non-hermitian corr. matrix which is roughly double the work.
+  isHermitian=false #Assume corr-matrix is non-hermitian
+  if haskey(kwargs, :isHermitian) #Did the user explicitly request something?
+    isHermitian = kwargs.isHermitian #Honour users request
+  else
+    O1=op(Op1, s, 1)
+    O2=op(Op2, s, 1)
+    O1/=norm(O1)
+    O2/=norm(O2)
+    #We need to decide if O1 ∝ O2 or O1 ∝ O2^dagger allowing for some round off errors.
+    eps=1e-10
+    is_proprtional=norm(O1 - O2)<eps;
+    is_hermitian  =norm(O1 - dag(swapprime(O2, 0, 1)))<eps;
+    if is_proprtional || is_hermitian 
+      isHermitian=true
+    end
+    # finally if they are both fermionic and proportional then the corr matrix will
+    # be anti symmetric insterad of Hermitian. Handle things like <C_i*C_j>
+    # at this point we know fermionic2=fermionic1, but we put them both in the if
+    # to clarify the meaning of what we are doing.
+    if is_proprtional && fermionic1 && fermionic2 
```

Comment:
Thanks for thinking through this carefully, to catch the antisymmetric fermionic case.